Bottom: e43d526611a8a81d627cb6ab4d6438aedbf4de41
Top:    14c76e2aa7170c03278e199a4cddbafb1d124482
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-03-06 00:48:57 +0000

Enable output of via pads, and dial down circular approximation accuracy


---

diff --git a/src/hid/common/object3d.c b/src/hid/common/object3d.c
index b00a1d2..20fa256 100644
--- a/src/hid/common/object3d.c
+++ b/src/hid/common/object3d.c
@@ -1212,6 +1212,31 @@ object3d_from_copper_layers_within_area (POLYAREA *area)
         }
 #endif
 
+      /* TODO: Inter-layer features
+       *
+       * Accumulate a circular polygon for each plated hole we may cut, ensuring
+       * the finished polygon contour extends to include the via barrel extents.
+       *
+       * Subtract non-plated hole contours from the polygons.
+       *
+       * For each hole, add the via-barrel between layers... removing the Object3D
+       * from the list of objects as they become joined with some other. (The final
+       * list of objects shuold match 1:1 with resultant bodies, and contain no
+       * duplicates.
+       *
+       * To accomodate overlapping drill holes, accumulate all via-barrels into a
+       * polygon, and subtract that from the positive copper polygon. As we already
+       * added via-barrels to each copper layer, the each barrel extrusion contour
+       * should match only one body of copper on a given layer.
+       *
+       * Remove the drilled hole down each via by extruding the additional faces.
+       *
+       * To accomodate overlapping drill holes, accumulate all drills into a polygon,
+       * and subtract that from the positive copper polygon. Any subtracted contour
+       * should at this point match to one body of copper on a given layer.
+       *
+       */
+
       if (info.poly == NULL)
         {
           fprintf (stderr, "Skipping layer group %i, info.poly was NULL\n", group);
